roles/ruby_s/tasks/main.yml

```diff
@@ -0,0 +1,88 @@
+---
+- name: ruby | check version
+  ansible.builtin.command: ruby -v
+  register: installed_ruby
+  ignore_errors: true
+  changed_when: false
+
+- name: ruby | install ruby if it-s not already there
+  block:
+  - name: ruby | check for ruby role apps
+    ansible.builtin.shell: dpkg-query -W 'ruby-switch'
+    ignore_errors: true
+    changed_when: false
+    register: switch
+
+  - name: ruby | uninstall global bundler
+    community.general.gem:
+      name: bundler
+      state: absent
+    when: switch is success
+
+  - name: ruby | remove installed ruby
+    ansible.builtin.apt:
+      name: ["{{ ubuntu_ruby_version }}", "{{ ubuntu_ruby_version }}-dev", "ruby-switch"]
+      state: absent
+      update_cache: true
+    when: switch is success
+
+  - name: rvm | remove the Brightbox repository
+    ansible.builtin.apt_repository:
+      repo: "ppa:brightbox/ruby-ng"
+      state: absent
+    when: switch is success
+
+  - name: ruby |lookup download path and checksum from ruby-lang release index
+    ansible.builtin.shell: curl https://cache.ruby-lang.org/pub/ruby/index.txt | grep {{ ruby_version }}.tar.gz
+    register: ruby_index_line
+
+  - name: ruby | separate index entry values
+    ansible.builtin.set_fact:
+      ruby_values: "{{ ruby_index_line.stdout | regex_findall('(\\S+)') }}"
+
+  - name: get ruby url
+    ansible.builtin.set_fact:
+      ruby_gzip_url: "{{ ruby_values[1] }}"
+
+  - name: get ruby checksum
+    set_fact:
+      ruby_sha_256: "{{ ruby_values[3] }}"
+
+  - name: ruby | setup install directory
```

Comment:
I might set this value in `defaults/main.yml` instead of using `set_fact` here. It's easier to override the role default.

Reply:
we get this dynamically by `curl https://cache.ruby-lang.org/pub/ruby/index.txt` and getting the values. So I don't think it is easily possible. You?